src/ir/auth_logic/ast.h

```diff
@@ -337,12 +354,19 @@ class Query {
   }
 
   // A potentially ugly print of the state in this class
-  // for debugging/testing only
+  // for debugging only
   std::string DebugPrint() const {
     return absl::StrCat("Query(", name_, principal_.DebugPrint(),
                         fact_.DebugPrint(), ")");
   }
 
+  bool operator==(const Query& rhs) const {
+    return name_ == rhs.name() && principal_ == rhs.principal() &&
```

Comment:
refer to fields directly to make the fieldwise comparison obvious. (Please change other implementations too.)

Reply:
fixed